### ISSUE TITLE
Pydantic Field alias - wrong invalid string reporting

### ISSUE THREAD

**Issue by yann-combarnous** [microsoft/pyright#9386]
**Describe the bug**
For Pydantic field alias, valid strings are reported as being invalid.

**Code or Screenshots**
```
class Resource(BaseModel):
    odataType: str = Field(alias="@odata.type")  # type: ignore[reportGeneralTypeIssues]
```

alias is reported as `str | None`, but pyright raises `reportGeneralTypeIssues`

pydantic = "2.9.2"

**VS Code extension or command-line**
Using VSCode Pylance, also reported using jakebailey/pyright-action, using pylance-version: latest.

This is a new issue, was not reported prior to last week.

**Comment by Viicos** [comment]
In Pydantic, you can also construct a model using the `model_validate_*` classmethods:

```python
class Resource(BaseModel):
    odataType: str = Field(alias="@odata.type")

resource = Resource.model_validate({'@odata.type': '...'})
```

Where it makes sense to use invalid Python identifiers as keys.

Here are a couple options that could be considered:
- Do not error when the alias is not a valid identifier. After all, users can still do `Resource(**{'@odata.type': '...'})`. This is similar to unpacked typed dictionaries, where the following currently works with no pyright error ([playground](https://pyright-play.net/?pythonVersion=3.12&strict=true&locale=en-us&enableExperimentalFeatures=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjADRQCqKCAhlQNYBQ3B5UAvIRIVqMABQByXhPoBvCQAEwpFjBYA6eAmISAXFADOMEAF8AlN06liwKMACuKKmIBUL9gHcWINAf1NWDgBtXgBdM311KMsHJ1cXeSUVNU0SPSgJKPUJc04gA)):

```python
from typing import TypedDict, Unpack


TD = TypedDict('TD', {'@odata.type': str})


def func(**kwargs: Unpack[TD]): ...


func(**{'@odata.type': '...'})
```
- Add a new kind of error so that these can easily be ignored at the project level. However, I don't know what's the policy when it comes to adding new/changin error codes in pyright.

@yann-combarnous: alternatively, you can set the `Field` as `Annotated` metadata. Pyright won't error, but won't be able to pick up other `Field` arguments, such as default values, etc:

```python
from typing import Annotated

class Resource(BaseModel):
    odataType: Annotated[str, Field(alias="@odata.type")] = "a"
    # Note that pyright won't be able to understand that the following field
    # has a default value:
    test: Annotated[str, Field(default="a")]

Resource()  # pyright error, missing value for argument `test`
```

**Comment by erictraut** [comment]
Oh, that's a good point about using an unpacked dictionary (as in `Resource(**{'@odata.type': '...'})`). 

@debonte, based on this feedback, I think I will undo the change I made for https://github.com/microsoft/pyright/issues/9220. This isn't an error condition unless/until you attempt to use the illegal identifier as as keyword argument. Pyright pyright already flags that as an error. The error check I added for #9220 is therefore a false positive.

If I revert the change for #9220, the completion provider logic will then need to become smarter to avoid recommending an illegal keyword argument identifier.

Here's a similar case — involving a TypedDict rather than a dataclass_transform class — where the completion provider produces problematic completion suggestions:

```python
TD1 = TypedDict("TD1", {"@illegal param": int})
td1 = TD1(**{"@illegal param": 3})
```

**Comment by debonte** [comment]
Sounds good, thanks.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.388.
